src/lib/utils/SynapseClient.tsx

```diff
@@ -608,3 +638,245 @@ export const signOut = () => {
     console.error('err when clearing the session cookie ', err)
   })
 }
+
+/**
+ * Upload file.  Note that this currently only supports Synapse storage (Sage s3 bucket)
+ * @param sessionToken
+ * @param file
+ * @param endpoint
+ */
+export const uploadFile = (
+  sessionToken: string | undefined,
+  filename: string,
+  file: Blob,
+  endpoint: string = DEFAULT_ENDPOINT,
+) => {
+  return new Promise((fileUploadResolve, fileUploadReject) => {
+    const partSize: number = Math.max(5242880, (file.size / 10000))
+    const request: MultipartUploadRequest = {
+      contentType: file.type,
+      fileName: filename,
+      fileSizeBytes: file.size,
+      partSizeBytes: partSize,
+      storageLocationId: SYNAPSE_STORAGE_LOCATION_ID
+    }
+    calculateMd5(file).then((md5: string) => {
+      request.contentMD5Hex = md5
+      startMultipartUpload(sessionToken, filename, file, request, fileUploadResolve, fileUploadReject, endpoint)
+    })
+  })
+}
+
+const calculateMd5 = (
+  fileBlob: File | Blob
+) => {
+  const bmf = new browserMd5File()
+  return new Promise((resolve, reject) => {
+    bmf.md5(
+      fileBlob,
+      (error: any, md5: string) => {
+        if (md5) {
+          resolve(md5)
+        } else if (error) {
+          reject(error)
+        }
+      },
+      (progress: number) => {
+        // console.log('progress: ', progress)
+      }
+    )
+  })
+}
+
+const processFilePart = (
+  partNumber: number,
+  multipartUploadStatus: MultipartUploadStatus,
+  sessionToken: string | undefined,
+  fileName: string,
+  file: Blob,
+  request: MultipartUploadRequest,
+  fileUploadResolve: (fileUpload: FileUploadComplete) => void,
+  fileUploadReject: (reason: any) => void,
+  endpoint: string = DEFAULT_ENDPOINT
+) => {
+  if (multipartUploadStatus.clientSidePartsState![partNumber - 1]) {
+    // no-op. this part has already been processed!
+    return
+  }
+
+  const uploadID = multipartUploadStatus.uploadId
```

Comment:
this could get renamed to upload**Id** and then in presignedUploadUrlRequest you could use the shorthand for uploadId, so 
{
contentType: 'application/octet-stream', // each part is binary
partNumbers: [partNumber],
uploadId
}